components/sync15/src/sync_multiple.rs

```diff
@@ -110,6 +120,19 @@ fn do_sync_multiple(
         sync_result.service_status = ServiceStatus::Interrupted;
         return Ok(());
     }
+
+    if match mem_cached_state.last_hmac {
+        // Wiping state when None costs nothing as there's is other state.
+        None => true,
+        // XXX - Does this do what I think it does? :) What's a better way?
+        Some(bytes) => !root_sync_key.verify_hmac(&bytes, "")?,
```

Comment:
The `verify_hmac` method is going to go away as part of a broader crypto refactor, but we could change this as part of that work as well. I think it'll do what you want, but I'd suggest hmacing a fixed non-empty string (like "sync_multiple in-memory cached state v1" or whatever) just purely for "hygiene" reasons. 

Are we able to get access to the `kid` of the key from FxA here (which is the same thing as X-Client-State`? It would probably suffice to use the `kid` for this comparison instead of hmacing anything.